executor/runtime/docker/docker.go

```diff
@@ -1935,17 +1968,17 @@ func (r *DockerRuntime) setupGPU(ctx context.Context, c *runtimeTypes.Container,
 }
 
 // Kill uses the Docker API to terminate a container and notifies the VPC driver to tear down its networking
-func (r *DockerRuntime) Kill(c *runtimeTypes.Container) error { // nolint: gocyclo
-	log.Infof("Killing %s", c.TaskID)
+func (r *DockerRuntime) Kill(ctx context.Context) error { // nolint: gocyclo
+	log.Infof("Killing %s", r.c.TaskID)
 
 	var errs *multierror.Error
 
-	containerStopTimeout := time.Second * time.Duration(c.TitusInfo.GetKillWaitSeconds())
+	containerStopTimeout := time.Second * time.Duration(r.c.TitusInfo.GetKillWaitSeconds())
 	if containerStopTimeout == 0 {
 		containerStopTimeout = defaultKillWait
 	}
 
-	if containerJSON, err := r.client.ContainerInspect(context.TODO(), c.ID); docker.IsErrNotFound(err) {
+	if containerJSON, err := r.client.ContainerInspect(context.TODO(), r.c.ID); docker.IsErrNotFound(err) {
```

Comment:
Can probably change `context.TODO()` to `ctx` here now?

Reply:
No, we can't, because by the time that this part (Kill) gets called, the parent context may already have been cancelled (hence why we're killing. I don't yet know how to solve this problem. Suggestions welcome.

Reply:
OK, I see.  The only thing I can think of is doing a `context.WithTimeout()`, so at least this entire block has some timeout associated with it.  Not the greatest, though.